### ISSUE TITLE
Java Exception: java.lang.IllegalStateException: Failed to recover - encoded with PDF_417 in 110x166

### ISSUE THREAD

**Issue by Microsvuln** [zxing/zxing#1427]
Hi .

Using jazzer, the following exception has been occurred :

`Java Exception: java.lang.IllegalStateException: Failed to recover` .

The produced jazzer file is as :
```

import java.lang.reflect.InvocationTargetException;
import java.lang.reflect.Method;
public class Crash_554492f3abaea0e383717bff996d44d0cdb12f5f {
    static final String base64Bytes = "rO0ABXNyABNqYXZhLnV0aWwuQXJyYXlMaXN0eIHSHZnHYZ0DAAFJAARzaXpleHAAAAADdwQAAAADc3IAEWphdmEubGFuZy5JbnRlZ2VyEuKgpPeBhzgCAAFJAAV2YWx1ZXhyABBqYXZhLmxhbmcuTnVtYmVyhqyVHQuU4IsCAAB4cAAAAG5zcQB+AAIAAACAdAAAeA==";

    public static void main(String[] args) {
        ClassLoader.getSystemClassLoader().setDefaultAssertionStatus(true);
        try {
            Method fuzzerInitialize = MultiFormatEncodeFuzzer.class.getMethod("fuzzerInitialize");
            fuzzerInitialize.invoke(null);
        } catch (NoSuchMethodException ignored) {
            try {
                Method fuzzerInitialize = MultiFormatEncodeFuzzer.class.getMethod("fuzzerInitialize", String[].class);
                fuzzerInitialize.invoke(null, (Object) args);
            } catch (NoSuchMethodException ignored) {
            } catch (IllegalAccessException | InvocationTargetException e) {
                e.printStackTrace();
                System.exit(1);
            }
        } catch (IllegalAccessException | InvocationTargetException e) {
            e.printStackTrace();
            System.exit(1);
        }
        com.code_intelligence.jazzer.api.CannedFuzzedDataProvider input = new com.code_intelligence.jazzer.api.CannedFuzzedDataProvider(base64Bytes);
        MultiFormatEncodeFuzzer.fuzzerTestOneInput(input);
    }
}


```
And the corpus required to reproduce (Please first unzip it) :

[crash-corpus.zip](https://github.com/zxing/zxing/files/7092192/crash-corpus.zip)

To reproduce :

`./MultiFormatEncodeFuzzer  [crashing-corpus]
`
Thanks!

**Comment by srowen** [comment]
Is this related to what the oss-fuzz project is finding? 
I see https://github.com/google/oss-fuzz/blob/90dc42fb873d18c515056d1577cd1ec5c291c4e5/projects/zxing/MultiFormatEncodeFuzzer.java but that doesn't seem to have a main method to run.

Or is it possible to see more of the stack trace? for many of these that's enough to guess what the problem is.

**Comment by Microsvuln** [comment]
Yes, this is actually true . It is a harness on the oss-fuzz project for zxing .

About the method, there is no problem that it does not have any main method. but it actually wraps the main method with fuzzerTestOneInput as it's the main method that is used by the jazzer fuzzer to test libraries . it actually uses zxing library as you can see and it goes to fuzz specific buffers in the program to find the bugs . Indeed the jazzer is the equivalent of LLVM's libfuzzer for java programs. 

The whole stack trace is as : 

```
OpenJDK 64-Bit Server VM warning: Sharing is only supported for boot loader classes because bootstrap classpath has been appended
INFO: Loaded 8 hooks from com.code_intelligence.jazzer.sanitizers.Deserialization
INFO: Loaded 1 hooks from com.code_intelligence.jazzer.sanitizers.ReflectiveCall
INFO: Loaded 8649 no-throw method signatures
INFO: Instrumented MultiFormatEncodeFuzzer (took 75 ms, size +40%)
INFO: Instrumented com.google.zxing.WriterException (took 3 ms, size +44%)
INFO: Instrumented com.google.zxing.NotFoundException (took 2 ms, size +50%)
INFO: Instrumented com.google.zxing.ReaderException (took 3 ms, size +39%)
INFO: Instrumented com.google.zxing.ChecksumException (took 3 ms, size +66%)
INFO: Instrumented com.google.zxing.FormatException (took 3 ms, size +66%)
INFO: Instrumented com.google.zxing.Binarizer (took 2 ms, size +31%)
INFO: Instrumented MultiFormatEncodeFuzzer$TrivialBinarizer (took 3 ms, size +34%)
INFO: Instrumented com.google.zxing.Reader (took 1 ms, size +0%)
INFO: libFuzzer ignores flags that start with '--'
INFO: Running with entropic power schedule (0xFF, 100).
INFO: Seed: 2576755031
INFO: Loaded 1 modules   (512 inline 8-bit counters): 512 [0x7f2a3bcfc010, 0x7f2a3bcfc210), 
INFO: Loaded 1 PC tables (512 PCs): 512 [0x7f2a3a8fa010,0x7f2a3a8fc010), 
./jazzer_driver: Running 1 inputs 1 time(s) each.
Running: crash-c27af73e7dfe7bd099a506c73fad071a4c01f540
INFO: Instrumented com.google.zxing.BarcodeFormat (took 5 ms, size +41%)
INFO: Instrumented com.google.zxing.MultiFormatWriter (took 4 ms, size +34%)
INFO: Instrumented com.google.zxing.Writer (took 1 ms, size +0%)
INFO: Instrumented com.google.zxing.MultiFormatWriter$1 (took 3 ms, size +91%)
INFO: Instrumented com.google.zxing.pdf417.PDF417Writer (took 7 ms, size +48%)
INFO: Instrumented com.google.zxing.pdf417.encoder.PDF417 (took 27 ms, size +7%)
INFO: Instrumented com.google.zxing.pdf417.encoder.Compaction (took 2 ms, size +41%)
INFO: Instrumented com.google.zxing.pdf417.encoder.PDF417ErrorCorrection (took 7 ms, size +11%)
INFO: New number of inline 8-bit counters: 1024
INFO: Instrumented com.google.zxing.pdf417.encoder.PDF417HighLevelEncoder (took 13 ms, size +73%)
INFO: Instrumented com.google.zxing.pdf417.encoder.PDF417HighLevelEncoder$1 (took 1 ms, size +52%)
INFO: Instrumented com.google.zxing.pdf417.encoder.BarcodeMatrix (took 3 ms, size +46%)
INFO: Instrumented com.google.zxing.pdf417.encoder.BarcodeRow (took 2 ms, size +64%)
INFO: Instrumented com.google.zxing.common.BitMatrix (took 12 ms, size +71%)
INFO: Instrumented com.google.zxing.BinaryBitmap (took 2 ms, size +43%)
INFO: Instrumented com.google.zxing.LuminanceSource (took 2 ms, size +39%)
INFO: Instrumented MultiFormatEncodeFuzzer$TrivialLuminanceSource (took 1 ms, size +95%)
INFO: Instrumented com.google.zxing.InvertedLuminanceSource (took 3 ms, size +57%)
INFO: Instrumented MultiFormatEncodeFuzzer$1 (took 2 ms, size +95%)
INFO: New number of inline 8-bit counters: 2048
INFO: Instrumented com.google.zxing.pdf417.PDF417Reader (took 4 ms, size +35%)
INFO: Instrumented com.google.zxing.multi.MultipleBarcodeReader (took 0 ms, size +0%)
INFO: Instrumented com.google.zxing.Result (took 2 ms, size +32%)
INFO: Instrumented com.google.zxing.pdf417.detector.Detector (took 4 ms, size +58%)
INFO: Instrumented com.google.zxing.ResultPoint (took 2 ms, size +50%)
INFO: Instrumented com.google.zxing.common.BitArray (took 13 ms, size +65%)
INFO: Instrumented com.google.zxing.pdf417.detector.PDF417DetectorResult (took 1 ms, size +26%)
INFO: Instrumented com.google.zxing.pdf417.decoder.PDF417ScanningDecoder (took 10 ms, size +70%)
INFO: Instrumented com.google.zxing.pdf417.decoder.DetectionResultColumn (took 2 ms, size +51%)
INFO: Instrumented com.google.zxing.pdf417.decoder.DetectionResultRowIndicatorColumn (took 4 ms, size +80%)
INFO: New number of inline 8-bit counters: 4096
INFO: Instrumented com.google.zxing.pdf417.decoder.ec.ErrorCorrection (took 3 ms, size +64%)
INFO: Instrumented com.google.zxing.pdf417.decoder.ec.ModulusGF (took 1 ms, size +46%)
INFO: Instrumented com.google.zxing.pdf417.decoder.ec.ModulusPoly (took 3 ms, size +72%)
INFO: Instrumented com.google.zxing.pdf417.decoder.BoundingBox (took 2 ms, size +62%)
INFO: Instrumented com.google.zxing.pdf417.decoder.Codeword (took 1 ms, size +46%)
INFO: Instrumented com.google.zxing.common.detector.MathUtils (took 1 ms, size +52%)
INFO: Instrumented com.google.zxing.pdf417.decoder.PDF417CodewordDecoder (took 2 ms, size +65%)
INFO: Instrumented com.google.zxing.pdf417.PDF417Common (took 16 ms, size +1%)
INFO: Instrumented com.google.zxing.pdf417.decoder.BarcodeValue (took 2 ms, size +44%)
INFO: Instrumented com.google.zxing.pdf417.decoder.BarcodeMetadata (took 1 ms, size +34%)
INFO: Instrumented com.google.zxing.pdf417.decoder.DetectionResult (took 3 ms, size +66%)
INFO: Instrumented com.google.zxing.pdf417.decoder.DecodedBitStreamParser (took 8 ms, size +67%)
INFO: Instrumented com.google.zxing.pdf417.PDF417ResultMetadata (took 1 ms, size +29%)

== Java Exception: java.lang.IllegalStateException: Failed to recover

encoded with PDF_417 in 110x166

                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                                                                                                                                
                                                                                                                                                                
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                                                                                                                                
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                                                                                                                                
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                                                                                                                                
                                                                            X X X X                                                                             
                                                                            X X X X                                                                             
                                                                    X X X X X X X X X X X X                                                                     
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X         X X X X X X X X                                                                     
                                                            X X X X X X X X                                                                                     
                                                                                            X X X X                                                             
                                                            X X X X X X X X         X X X X X X X X                                                             
                                                                            X X X X X X X X X X X X                                                             
                                                            X X X X X X X X         X X X X                                                                     
                                                                    X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X                                                                                     
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X         X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                                                                                                                                
                                                                    X X X X                                                                                     
                                                            X X X X                                                                                             
                                                                    X X X X                                                                                     
                                                                            X X X X                                                                             
                                                            X X X X         X X X X X X X X                                                                     
                                                            X X X X X X X X X X X X         X X X X                                                             
                                                                    X X X X X X X X                                                                             
                                                                    X X X X         X X X X X X X X                                                             
                                                                            X X X X                                                                             
                                                            X X X X                         X X X X                                                             
                                                            X X X X X X X X                                                                                     
                                                                    X X X X X X X X X X X X                                                                     
                                                                    X X X X         X X X X X X X X                                                             
                                                                    X X X X                 X X X X                                                             
                                                                    X X X X         X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                                                                                                                                
                                                                            X X X X X X X X X X X X                                                             
                                                            X X X X         X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X                                                                     
                                                            X X X X X X X X X X X X                                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X                 X X X X                                                             
                                                                                            X X X X                                                             
                                                            X X X X X X X X                 X X X X                                                             
                                                                            X X X X         X X X X                                                             
                                                            X X X X                 X X X X X X X X                                                             
                                                                    X X X X         X X X X                                                                     
                                                            X X X X X X X X X X X X         X X X X                                                             
                                                            X X X X X X X X         X X X X X X X X                                                             
                                                            X X X X X X X X         X X X X                                                                     
                                                            X X X X         X X X X X X X X                                                                     
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                                                                                                                                
                                                                    X X X X                                                                                     
                                                                    X X X X                 X X X X                                                             
                                                                    X X X X X X X X X X X X X X X X                                                             
                                                            X X X X         X X X X X X X X                                                                     
                                                            X X X X         X X X X X X X X                                                                     
                                                            X X X X         X X X X X X X X                                                                     
                                                            X X X X X X X X X X X X                                                                             
                                                                            X X X X         X X X X                                                             
                                                            X X X X X X X X         X X X X X X X X                                                             
                                                                            X X X X X X X X X X X X                                                             
                                                            X X X X X X X X         X X X X                                                                     
                                                                    X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X                                                                                     
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X         X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                                                                                                                                
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                                                                                                                                
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                                                                                                                                
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                


                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                                                                                                                                
                                                                                                                                                                
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                                                                                                                                
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                                                                                                                                
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                                                                                                                                
                                                                            X X X X                                                                             
                                                                            X X X X                                                                             
                                                                    X X X X X X X X X X X X                                                                     
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X         X X X X X X X X                                                                     
                                                            X X X X X X X X                                                                                     
                                                                                            X X X X                                                             
                                                            X X X X X X X X         X X X X X X X X                                                             
                                                                            X X X X X X X X X X X X                                                             
                                                            X X X X X X X X         X X X X                                                                     
                                                                    X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X                                                                                     
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X         X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                                                                                                                                
                                                                    X X X X                                                                                     
                                                            X X X X                                                                                             
                                                                    X X X X                                                                                     
                                                                            X X X X                                                                             
                                                            X X X X         X X X X X X X X                                                                     
                                                            X X X X X X X X X X X X         X X X X                                                             
                                                                    X X X X X X X X                                                                             
                                                                    X X X X         X X X X X X X X                                                             
                                                                            X X X X                                                                             
                                                            X X X X                         X X X X                                                             
                                                            X X X X X X X X                                                                                     
                                                                    X X X X X X X X X X X X                                                                     
                                                                    X X X X         X X X X X X X X                                                             
                                                                    X X X X                 X X X X                                                             
                                                                    X X X X         X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                                                                                                                                
                                                                            X X X X X X X X X X X X                                                             
                                                            X X X X         X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X                                                                     
                                                            X X X X X X X X X X X X                                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X                 X X X X                                                             
                                                                                            X X X X                                                             
                                                            X X X X X X X X                 X X X X                                                             
                                                                            X X X X         X X X X                                                             
                                                            X X X X                 X X X X X X X X                                                             
                                                                    X X X X         X X X X                                                                     
                                                            X X X X X X X X X X X X         X X X X                                                             
                                                            X X X X X X X X         X X X X X X X X                                                             
                                                            X X X X X X X X         X X X X                                                                     
                                                            X X X X         X X X X X X X X                                                                     
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                                                                                                                                
                                                                    X X X X                                                                                     
                                                                    X X X X                 X X X X                                                             
                                                                    X X X X X X X X X X X X X X X X                                                             
                                                            X X X X         X X X X X X X X                                                                     
                                                            X X X X         X X X X X X X X                                                                     
                                                            X X X X         X X X X X X X X                                                                     
                                                            X X X X X X X X X X X X                                                                             
                                                                            X X X X         X X X X                                                             
                                                            X X X X X X X X         X X X X X X X X                                                             
                                                                            X X X X X X X X X X X X                                                             
                                                            X X X X X X X X         X X X X                                                                     
                                                                    X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X                                                                                     
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X         X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                                                                                                                                
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                                                                                                                                
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                                                                                                                                
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                            X X X X X X X X X X X X X X X X X X X X                                                             
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                
                                                                                                                                                                

        at MultiFormatEncodeFuzzer.fuzzerTestOneInput(MultiFormatEncodeFuzzer.java:89)
Caused by: com.google.zxing.FormatException
DEDUP_TOKEN: 4abbca5668ca9e95
== libFuzzer crashing input ==
reproducer_path='.'; Java reproducer written to ./Crash_c27af73e7dfe7bd099a506c73fad071a4c01f540.java

```
Let me know if you have other questions .

**Comment by srowen** [comment]
Oh I see this is just https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=37794&q=label%3AProj-zxing 
Yeah, this one I'm having a harder time figuring out. It's a real issue I'm pretty sure, even if a corner case.

**Comment by Microsvuln** [comment]
It seems that I just don't have access to the link you have sent. this link is only for googlers or project maintainers who have their project under OSS-Fuzz .

Then if you think this is the same problem as I reported here, so this is a duplicate (Based on dates), unless, I think this should be addressed separately.

Thanks!

- Microsvuln

**Comment by srowen** [comment]
Looks like it's not public, yeah. It could well be the same. The oss-fuzz project is running fuzzing automatically now and is generating a number of issues like this. This looks like one I saw a few days ago and I'm not sure how to fix it without a fair bit more work.

**Comment by pleleraunya** [comment]
Ok

**Comment by Ranchosm335** [comment]
A here so thin you can blow through it burn baby burn

**Comment by AlexGeller1** [comment]
I had a stab at this and I think that it might be that the PDF417 decoder raises an error when the there is nothing to decode (some of the decoders including this one do that). If I am right, then there is only one codeword (900 -> switch to text mode) to decode. I am a little bit unsure because from your debug output I can't really tell. I resorted to parsing the matrix output but to really be sure, could you change line 88 in MultiFormatEncodeFuzzer.java from
```java
throw new IllegalStateException("Failed to recover\n" + originalData + "\nencoded with " + format + " in "
                    + width + "x" + height + "\n\n" + matrix.toString() + "\n\n" + bitmap.toString(), e);
```
to
```java
throw new IllegalStateException("Failed to recover\n" + insertEscapes(originalData) + "\nencoded with " + format + " in "
                    + width + "x" + height + "\n\n" + matrix.toString() + "\n\n" + bitmap.toString(), e);
```
and add the function insertEscapes:
```java
private static String insertEscapes(String s) {
    StringBuilder sb = new StringBuilder();
    for (int i = 0; i < s.length(); i++) {
        char ch = s.charAt(i);
        switch (ch) {
            case '\b':  sb.append("\\b"); break;
            case '\f' : sb.append("\\f"); break;
            case '\n' : sb.append("\\n"); break;
            case '\r' : sb.append("\\r"); break;
            case '\t' : sb.append("\\t"); break;
            default : {
                if(ch >= 32 && ch <= 127) {
                  sb.append(ch); 
                } else {
                  sb.append(String.format("\\u%04X", (int)ch));
                }
            }
        }
    } 
    return sb.toString();
  }
```

**Comment by AlexGeller1** [comment]
@srowen: The fuzz test is likely failing trying to encode the empty string. The exception in this case is raised in DecodedBitStreamParser in line 148 (see https://github.com/zxing/zxing/blob/master/core/src/main/java/com/google/zxing/pdf417/decoder/DecodedBitStreamParser.java#L148)
I already stumbled on this earlier when I had to exclude the empty string test from one of the new encode/decode tests (see https://github.com/zxing/zxing/pull/1504/commits/ddbe649975a86395ff6ef9ea72ee95598d5b269e#diff-447fb1fe6ea758a8e9fc3f4c1c8c7eaa3aa3a403a7df35b31779345fe2e19400R186).
We could remove that check. What do you think?

**Comment by srowen** [comment]
Doesn't seem like the empty string - there is a whole barcode here in the log output. It failed to decode something that was encoded

**Comment by AlexGeller1** [comment]
I decoded the matrix and the codewords are 2, 900, 792, 768, 189, 329, 324, 285, 547, 210.
The first codeword is the length so that this means that it encodes solely the value 900 which is the switch to text mode.
Sure, it could be that I have made some sort of mistake (that's why I asked Microsvuln for better output) but I think that we should be prepared that that might be the cause of the problem)

**Comment by srowen** [comment]
OK then a problem with the encoding? I think this test is a round-trip one

**Comment by AlexGeller1** [comment]
No, the encoder is fine. When you encode the empty string then it produces codeword 900. The encoder always prefixes 900 even though that's unnecessary. It is harmless though. If it didn't do that then it would fail the same way. Then the codewords would be "1, ....." and the decoder would fail at the same place.

**Comment by srowen** [comment]
Right, I mean it is probably not encoding the empty string, but something that results in this incorrect encoding - seems like it doesn't make sense that there are 'no' values but clearly there are more codewords?

**Comment by AlexGeller1** [comment]
The other codewords (those beyond the first two) are the error correction codewords.

**Comment by AlexGeller1** [comment]
More clearly, the first value denotes the length of the data including the length value. In this case it is 2 so that the data is declared as being of length 1. So 900 is the data. The codewords behind that are error correction and perhaps padding.

**Comment by srowen** [comment]
I think that's ok. This test is asserting that what can be encoded can be decided, not that everything can be encoded

**Comment by gredler** [comment]
@srowen Can this be closed? It looks like PR #1523 is merged.
